android/sdk/src/main/java/com/taobao/weex/dom/BasicEditTextDomObject.java

```diff
@@ -0,0 +1,289 @@
+ *
+ *    Copyright 2016 Alibaba Group
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.taobao.weex.dom;
+
+import android.text.BoringLayout;
+import android.text.Layout;
+import android.text.TextPaint;
+import android.util.TypedValue;
+import com.taobao.weex.common.WXDomPropConstant;
+import com.taobao.weex.dom.flex.CSSConstants;
+import com.taobao.weex.dom.flex.CSSNode;
+import com.taobao.weex.dom.flex.MeasureOutput;
+import com.taobao.weex.dom.flex.Spacing;
+import com.taobao.weex.ui.component.WXText;
+import com.taobao.weex.utils.WXViewUtils;
+
+import java.util.Map;
+
+import static com.taobao.weex.dom.WXStyle.UNSET;
+
+/**
+ * Created by sospartan on 7/12/16.
+ */
+public class BasicEditTextDomObject extends WXDomObject {
+
+  private TextPaint mPaint = new TextPaint();
+
+  private int mFontSize = UNSET;
+  private int mLineHeight = UNSET;
+  private String mFontFamily = null;
+
+  public BasicEditTextDomObject(){
+    super();
+    mPaint.setTextSize(WXViewUtils.getRealPxByWidth(WXText.sDEFAULT_SIZE));
+    setMeasureFunction(new MeasureFunction() {
+      @Override
+      public void measure(CSSNode node, float width, MeasureOutput measureOutput) {
+        if (CSSConstants.isUndefined(width)) {
+          width = node.cssstyle.maxWidth;
+        }
+        if (mLineHeight != UNSET && mLineHeight >= 0) {
+          measureOutput.height = mLineHeight;
+        } else {
+          measureOutput.height = getMeasureHeight();
+        }
+        measureOutput.width = width;
+      }
+    });
+  }
+
+  @Override
+  public void layoutBefore() {
+    super.layoutBefore();
+    updateStyleAndAttrs();
+  }
+
+  protected final float getMeasuredLineHeight(){
+    return mPaint.getFontMetrics(null);
+  }
+
+  protected float getMeasureHeight(){
+    Spacing padding = getPadding();
+    return getMeasuredLineHeight();
+  }
+
+  protected void updateStyleAndAttrs(){
+    if (style != null) {
+      if (style.containsKey(WXDomPropConstant.WX_FONTSIZE)) {
+        mFontSize = WXStyle.getFontSize(style);
+      }
+
+      if (style.containsKey(WXDomPropConstant.WX_FONTFAMILY)) {
+        mFontFamily = WXStyle.getFontFamily(style);
+      }
+
+      int lineHeight=WXStyle.getLineHeight(style);
+      if(lineHeight!=UNSET)
+        mLineHeight=lineHeight;
+
+      mPaint.setTextSize(mFontSize);
+      mPaint.setTypeface(WXCustomStyleSpan.getOrCreateTypeface(mFontFamily,mFontSize));
```

Comment:
font-weight and font-style should be considered. Reference the following code snippet.
```
  private static void apply(Paint paint, int style, int weight, String family) {
    int oldStyle;
    Typeface typeface = paint.getTypeface();
    if (typeface == null) {
      oldStyle = 0;
    } else {
      oldStyle = typeface.getStyle();
    }

    int want = 0;
    if ((weight == Typeface.BOLD)
        || ((oldStyle & Typeface.BOLD) != 0 && weight == WXStyle.UNSET)) {
      want |= Typeface.BOLD;
    }

    if ((style == Typeface.ITALIC)
        || ((oldStyle & Typeface.ITALIC) != 0 && style == WXStyle.UNSET)) {
      want |= Typeface.ITALIC;
    }

    if (family != null) {
      typeface = getOrCreateTypeface(family, want);
    }

    if (typeface != null) {
      paint.setTypeface(Typeface.create(typeface, want));
    } else {
      paint.setTypeface(Typeface.defaultFromStyle(want));
    }
  }
```